sceptre/config/reader.py

```diff
@@ -424,6 +439,29 @@ def _construct_stack(self, rel_path, stack_group_config=None):
         parsed_stack_group_config = self._parsed_stack_group_config(stack_group_config)
         config = self.read(rel_path, stack_group_config)
         stack_name = path.splitext(rel_path)[0]
+
+        for attribute in config:
+            if attribute not in VALID_KEYS:
+                self.logger.warning("Key {0} in stack {1} is unrecognised.".format(
+                    attribute, stack_name)
+                )
+
+        for required_key in STACK_GROUP_CONFIG_ATTRIBUTES.required:
+            if required_key not in config:
+                raise InvalidConfigFileError(
+                    "Required key '{0}' not found in Stack Group Config of '{1}'.".format(
+                        required_key, stack_name
+                    )
+                )
+
+        for required_key in STACK_CONFIG_ATTRIBUTES.required:
+            if required_key not in config:
+                raise InvalidConfigFileError(
+                    "Required key '{0}' not found in Stack Config ({1})".format(
```

Comment:
Probably want to keep the error message format consistent with the one above.

If we don't care about the _exact_ error message (Stack Config / Stack Group Config) then we should probably use a single loop to check both required config attributes.

I believe you mentioned that the **stack group config** can be missing with everything in the **stack config** instead, which means that the error message above (Stack Config / Stack Group Config) won't be correct anyhow. We could fix the underlying issue to force the correct attributes to exist in the `config.yaml` but that would definitely be a breaking change at this point, so I'd rather suggest we change the error message instead.